### ISSUE TITLE
[FEATURE] Migrate `GraderStore` to `pinia` instead of `vuex`

### ISSUE THREAD

**Issue by HNOONa-0** [omegaup/omegaup#7826]
The primary issue with `vuex` stores is the lack of TypeScript support. To address this, we can migrate to `pinia`, which offers TypeScript support out of the box and is actually re-branded Vuex5.

A full migration of all vuex stores is not necessary. Instead, we can start by migrating the `GraderStore` as a first step. This could also serve as a reference point for future migrations if we decide to migrate all stores.

**Comment by pabo99** [comment]
> I'd love to work on this issue! I'm familiar with Vuex and Pinia, and I'm currently contributing to related areas like the problem creator. Could I please be assigned to this?

Excellent!

How extensive do you think solving this issue could become? Would it be possible to break it down into smaller tasks? If that's feasible, we can start by creating sub-issues.

Also, taking advantage of the migration to Pinia, how viable do you think it would be to gradually migrate to Vue 3 in a way that isn't too painful?

**Comment by pushkardev123** [comment]
I believe the migration can be scoped into a few concrete sub-tasks:
1. Analyzing current usage of `GraderStore` and its modules.
2. Creating equivalent Pinia store(s) and ensuring backward compatibility.
3. Refactoring consuming components (e.g., `DiffEditor.vue`) to use the new store.
4. Writing unit tests to verify behavior hasn't regressed.

I’ll be happy to begin by drafting a migration plan and breaking the work into sub-issues.

Also, while working on the Markdown rendering refactor, I faced compatibility issues because I initially used Vue 3 syntax (e.g., `<script setup>`, Composition API), not realizing that the current version is still `vue@2.5.22`. 

Vue 3 introduces several breaking changes that would require a large-scale refactor of the entire codebase, including updates to dependencies, plugins, and testing infrastructure. Upgrading directly would introduce high risk and complexity. Adopting Vue 2.7 first allows us to gradually introduce modern features like the Composition API and <script setup> while maintaining compatibility with the existing Vue 2 code. This provides a safer and more manageable migration path toward Vue 3.

A possible first step could be updating to Vue 2.7, which adds Composition API and helps ease future Vue 3 migration.

Regarding Vue 3 migration, I think it’s viable to gradually move in parallel:
- Start with the Vue 2.7 compatibility build.
- Use the Composition API progressively via plugins.
- Upgrade components incrementally once Vue 3 is fully adopted.

Let me know your thoughts — I’ll start investigating and can open sub-issues accordingly.

**Comment by pabo99** [comment]
Go ahead with creating the sub-tasks!

Also, I like the idea of moving to Vue 2.7. Let us know if you need any help getting started with the migration (It can be nadled in a different issue)

**Comment by pushkardev123** [comment]
Thanks for the go-ahead! I’ll proceed with breaking down the work into sub-issues. Let me know if there’s anything else you’d like considered before moving forward.

And also for the pr #8143  I’ve resolved the requested changes and tested the functionality locally. Kindly requesting your review when time permits so we can move ahead. Thank you!

**Comment by pushkardev123** [comment]
Following up on the migration progress:

- ✅ Pinia version of `GraderStore` has been fully implemented, including typed state, getters, and actions.
- ✅ `DiffEditor.vue` has been refactored and is working correctly with the new store.
- ✅ Defensive logic has been added to key getters to avoid undefined runtime errors.
- ✅ Initial runtime validation confirms expected behavior.

To track and complete the full transition, I'm creating the following sub-issues:

1. Migrate all components using `GraderStore` to the Pinia store
2. Write unit tests for the new `GraderStore`
3. Clean up the legacy Vuex `GraderStore`

These will be handled progressively. Let me know if you'd like any of these prioritized differently.

A PR for the current migration will be opened shortly, which will include further testing and refinements as we progress with component integration.

**Comment by pushkardev123** [comment]
Quick update regarding this PR: while working on the migration, I encountered some issues related to the store during testing. Additionally, the overall scope of the changes is broader than I initially anticipated. Due to my university examinations, I will not be able to continue work immediately. I plan to resume and complete this migration after the first week of May.

**Comment by pabo99** [comment]
No worries at all!

 it's totally fine to take your time. Best of luck with your exams, and feel free to resume the work whenever you're ready after first week of May

**Comment by neerajbachani** [comment]
Hey @pabo99! I'd love to take on this issue.

I spent some time digging into the codebase before reaching out, so I have a pretty clear picture of what the migration involves.
We know the project is on Vue 2.7.16 which already has Composition API built in, so no extra setup needed there. Vuex will stay in place for the other stores like notificationsStore and mainStore - only GraderStore is moving to Pinia. Pinia itself isn't installed yet but adding 2.x is straightforward since it auto-detects Vue 2.7.

In total I'm looking at around 13 files - GraderStore.ts needs a full rewrite, util.ts needs its helper functions updated, 8 components across grader and graderv2 need to be updated, and one test file.

The trickiest part is that the current store uses dotted string keys like 'request.source' to access getters, which Pinia doesn't support. My plan is to add a small $get(key) / $set(key, value) helper inside the new Pinia store that acts as a compatibility layer for the old key format - this way component changes stay minimal.

I'd split this into two PRs - first one covers installing Pinia and rewriting the store and util.ts, second one handles all the component updates and the test file.

Could you assign this to me? Happy to chat through the approach if you'd like before I start!

**Comment by neerajbachani** [comment]
Hi @pabo99  I opened a **draft** PR for #7826 (GraderStore -> Pinia). This is **Part 1 only**: Pinia bootstrap (`grader/pinia.ts`), `useGraderStore` bridge to the existing Vuex store, no component changes yet. This should also make a later **Vue 3** migration a bit easier for the grader, since we’re aligning with Pinia early. I’ll mark it ready for review after the checklist and any guidance on one PR vs stacked PRs for the remaining parts.
